ORStools/gui/directions_gui.py

```diff
@@ -135,14 +103,50 @@ def get_parameters(self):
         if self.dlg.avoidpolygon_group.isChecked():
             layer = self.dlg.avoidpolygon_dropdown.currentLayer()
             if layer:
-                polygons = _get_avoid_polygons(layer)
+                polygons = self._get_avoid_polygons(layer)
                 self.options['avoid_polygons'] = polygons
 
         if self.options:
             params['options'] = self.options
 
         return params
 
+    def _get_avoid_polygons(self, layer):
+        """
+        Extract polygon geometries from the selected polygon layer.
+
+        :param layer: The polygon layer
+        :type layer: QgsMapLayer
+        :returns: GeoJSON object
+        :rtype: dict
+        """
+        polygons = None
+        transformer = transform.transformToWGS(layer.sourceCrs())
+        features = layer.getFeatures()
+
+        # iterate over all other features
+        for feature in features:
+            if feature.hasGeometry():
+                geom = feature.geometry()
+                geom.transform(transformer)
+                if polygons is None:
+                    polygons = geom
+                else:
+                    polygons = polygons.combine(geom)
+
+        if not polygons:
+            QMessageBox.warning(
```

Comment:
I think we should issue this warning box in `ORStoolsDialog` as QMessageBox is already imported there.
We can then just return polygons here.